customer-docs/user-manual.md

```diff
@@ -379,6 +385,34 @@ The Snowpipe Append template is A Snowflake data pipeline to append newly arrivi
 
 *NOTE:* `clean` does not remove the schema nor does it the `stage` which can be cleaned by `make drop-stage`.
 
+## QA Process 
+
+###Overview
+When creating data pipelines, measuring quality is always very important. Streamliner supports templates for collecting metrics
+on columns of tables and compare them over time. This method is based off of [control charts](https://en.wikipedia.org/wiki/Control_chart)
+to measure the variablility of the amount of data over time.
+
+###Process
+To take a control chart process and convert it to pipelines the process needs to complete the following steps:
+
+1.	Capture the occurrences of data values across all columns in a table.
+2.	Not capture data that is small or a small percentage of the values in a column.
+3.	Store each “run” of data in a metrics vault.
+4.	Compare the current “run” to the average and standard deviation of the previous runs for each value in each column of the table.
+
+Starting out with collecting and storing the data, in every database using this process we would want to create a “Metrics” table to store all of the data. That table would have the time and date of the run, the column name being looked at, the value in that column, and the count of records with that value. To collect the data in snowflake, a stored procedure can find all the columns on the table, calculate the counts for each value, and insert the data into the “Metrics” table.
+
+To easily collect the data, we can add a stream to the table we are monitoring and have a task that runs the stored procedure to collect the data. Once the data is collected and stored, the stored procedure can check if the values are within the UCL and LCL or 3 standard deviations away from the historical mean. If there are any values outside the range, the stored procedure can call an external api alerting users that some of the date requires investigation. Below is a diagram showing the dataflow.
+
+![QA_process](../images/streamliner_qa-process.png)
+
+####Dos and Don'ts
+This process should be used when monitoring on ongoing pipeline. It will not tell you if something is wrong or confirm data fits a predefined structure.
+If will give fail a check if the amount of data spikes up or down, if fields 
```

Comment:
Can you clarify the verbiage here? What does a failure look like or do? Does it send some alert or cause a process to exit uncleanly?

Reply:
@bpmcd I think the verbiage above would answer that question a bit more. I do agree that verbiage could be a bit clearer

> If there are any values outside the range, the stored procedure can call an external api alerting users that some of the date requires investigation.